x/staging/posts/keeper/keeper_reactions.go

```diff
@@ -13,86 +11,69 @@ import (
 // If another reaction from the same user for the same post and with the same value exists, returns an expError.
-func (k Keeper) SavePostReaction(ctx sdk.Context, postID string, reaction types.PostReaction) error {
+func (k Keeper) SavePostReaction(ctx sdk.Context, reaction types.PostReaction) error {
 	store := ctx.KVStore(k.storeKey)
-	key := types.PostReactionsStoreKey(postID)
-
-	// Get the existent reactions
-	var reactions types.PostReactions
-	k.cdc.MustUnmarshalBinaryBare(store.Get(key), &reactions)
 
+	key := types.PostReactionsStoreKey(reaction.PostID, reaction.Owner, reaction.ShortCode)
 	// Check for double reactions
-	if reactions.ContainsReactionFrom(reaction.Owner, reaction.ShortCode) {
+	if store.Has(key) {
 		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest,
 			"%s has already reacted with %s to the post with id %s",
-			reaction.Owner, reaction.ShortCode, postID)
+			reaction.Owner, reaction.ShortCode, reaction.PostID)
 	}
 
 	// Save the new reaction
-	store.Set(key, k.cdc.MustMarshalBinaryBare(&types.PostReactions{
-		Reactions: append(reactions.Reactions, reaction),
-	}))
-
+	store.Set(key, types.MustMarshalPostReaction(k.cdc, reaction))
 	return nil
 }
 
-// DeletePostReaction removes the reaction from the given user from the post having the
-// given postID. If no reaction with the same value was previously added from the given user, an expError
+// DeletePostReaction removes the reaction from the given data.
+// If no reaction with the same value was previously added from the given user, an expError
 // is returned.
-func (k Keeper) DeletePostReaction(ctx sdk.Context, postID string, reaction types.PostReaction) error {
+func (k Keeper) DeletePostReaction(ctx sdk.Context, reaction types.PostReaction) error {
 	store := ctx.KVStore(k.storeKey)
-	key := types.PostReactionsStoreKey(postID)
 
-	// Get the existing reactions
-	var wrapped types.PostReactions
-	k.cdc.MustUnmarshalBinaryBare(store.Get(key), &wrapped)
-
-	// Check if the user exists
-	reactions := types.NewPostReactions(wrapped.Reactions...)
-	if !reactions.ContainsReactionFrom(reaction.Owner, reaction.ShortCode) {
+	key := types.PostReactionsStoreKey(reaction.PostID, reaction.Owner, reaction.ShortCode)
+	if !store.Has(key) {
 		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest,
 			"cannot remove the reaction with value %s from user %s as it does not exist",
 			reaction.ShortCode, reaction.Owner)
 	}
 
-	// Remove and save the reactions list
-	if appendedReactions, edited := reactions.RemoveReaction(reaction.Owner, reaction.ShortCode); edited {
-		if len(appendedReactions.Reactions) == 0 {
-			store.Delete(key)
-		} else {
-			store.Set(key, k.cdc.MustMarshalBinaryBare(&appendedReactions))
-		}
-	}
-
+	store.Delete(key)
 	return nil
 }
 
 // GetPostReactions returns the list of reactions that has been associated to the post having the given id
 func (k Keeper) GetPostReactions(ctx sdk.Context, postID string) []types.PostReaction {
```

Comment:
I think this method can now be deleted since we are using the store prefix iterator to get the post reactions inside the gRPC method. Or maybe this is used somewhere else?

Reply:
It is used in `operation_reaction.go` to simply get post reactions in the post in 1 line.

Reply:
Since that's a test, I think we can move it inside that test directly without exposing it publicly to other modules